cmd/kube-proxy/app/server_others_test.go

```diff
@@ -278,17 +278,17 @@ func Test_getLocalDetector(t *testing.T) {
 			mode:        proxyconfigapi.LocalModeNodeCIDR,
 			config:      &proxyconfigapi.KubeProxyConfiguration{ClusterCIDR: "10.0.0.0/14"},
 			ipt:         utiliptablestest.NewIPv6Fake(),
-			expected:    nil,
+			expected:    proxyutiliptables.NewNoOpLocalDetector(),
 			nodeInfo:    makeNodeWithPodCIDRs("10.0.0.0/24"),
-			errExpected: true,
+			errExpected: false,
 		},
 		{
 			mode:        proxyconfigapi.LocalModeNodeCIDR,
 			config:      &proxyconfigapi.KubeProxyConfiguration{ClusterCIDR: "2002::1234:abcd:ffff:c0a8:101/64"},
 			ipt:         utiliptablestest.NewFake(),
-			expected:    nil,
+			expected:    proxyutiliptables.NewNoOpLocalDetector(),
 			nodeInfo:    makeNodeWithPodCIDRs("2002::1234:abcd:ffff:c0a8:101/96"),
-			errExpected: true,
+			errExpected: false,
```

Comment:
why should not we fail? this config will not work well

Reply:
To be clear, "this config" is "you have a host where IPv6 has been disabled at the kernel level, and then you try to add that host to a single-stack IPv6 cluster". And yeah, it will not work well. In fact, it will work so badly, I'm not convinced you could actually have gotten far enough to reach this error message...

If we really want to check this sort of stuff, then we should check _all_ of the relevant configuration ahead of time, and then fall back to single-stack mode if _any_ piece of configuration requires it, and error out if any two pieces of configuration are incompatible.

Reply:
(So, just to be clear: 99% of clusters run kube-proxy in dual-stack mode, because it only falls back to single-stack mode if IPv6 is disabled in the kernel. So changing the expected result of this test has no effect on 99% of users.)

Reply:
> single-stack mode if IPv6 is disabled in the kernel

I'm not sure the percentage is so great, some environments disable ipv6 for security reasons ... or for reasons

I think we should keep failing, maybe in another place we can validate and not here, but I don't think that being right 99% of times when we can be 100% is a good argument